py4DSTEM/process/diffraction/crystal_bloch.py

```diff
@@ -406,22 +418,26 @@ def generate_dynamical_diffraction_pattern(
 
     # calculate the diffraction intensities (and amplitudes) for each thichness matrix
     # I = |psi|^2 ; psi = C @ E(z) @ C^-1 @ psi_0, where E(z) is the thickness matrix
-    if return_complex: 
-  
-        # calculate the amplitudes 
+    if return_Smatrix:
+        Smatrices = [
+            C @ np.diag(np.exp(2.0j * np.pi * z * gamma)) @ C_inv
+            for z in np.atleast_1d(thickness)
+        ]
+        return Smatrices, psi_0
```

Comment:
Is it worth having the ability to generate the `pointlist` object as usual and then return  ~ `pointlist_obj, Smatricies_list, psi_0`, or spin this into a new function `generate_dynamical_Smatricies`  ?

Reply:
Someday in the future I do want to make the Bloch code more modular, in which case this will become a separate function, but I don't want to tackle that problem yet. For now, I think this way is clean enough.